core/src/al/miniaudio_context.cpp

```diff
@@ -0,0 +1,369 @@
+        : mContext(context)
+    {
+        ma_device_info* pPlaybackDeviceInfos;
+        ma_uint32 playbackDeviceCount;
+        ma_result result =
+            ma_context_get_devices(&mContext, &pPlaybackDeviceInfos, &playbackDeviceCount, nullptr, nullptr);
+
+        if (result != MA_SUCCESS)
+        {
+            CUBOS_FAIL("Failed to enumerate audio devices");
+            return;
+        }
+
+        ma_device_id* deviceId = nullptr;
+        for (ma_uint32 i = 0; i < playbackDeviceCount; i++)
+        {
+            if (deviceName == pPlaybackDeviceInfos[i].name)
+            {
+                deviceId = &pPlaybackDeviceInfos[i].id;
+                break;
+            }
+        }
+
+        if (deviceId == nullptr)
+        {
+            CUBOS_FAIL("Audio device '{}' not found", deviceName);
+            return;
+        }
+
+        ma_engine_config engineConfig = ma_engine_config_init();
+
+        if (listenerCount > MA_ENGINE_MAX_LISTENERS)
+        {
+            CUBOS_FAIL("Maximum number of listeners is 4");
+            return;
+        }
+
+        engineConfig.listenerCount = listenerCount;
+        engineConfig.pPlaybackDeviceID = deviceId; // Use the found device ID
+
+        if (ma_engine_init(&engineConfig, &mEngine) != MA_SUCCESS)
+        {
+            CUBOS_FAIL("Failed to initialize audio engine");
+            return;
+        }
+
+        mValid = true;
+
+        mListeners.reserve(listenerCount);
+        for (ma_uint32 i = 0; i < listenerCount; ++i)
+        {
+            mListeners.emplace_back(std::make_shared<MiniaudioListener>(mEngine, i));
+        }
+    }
+
+    ~MiniaudioDevice() override
+    {
+        ma_device_uninit(&mDevice);
+    }
+
+    Source createSource() override
+    {
+        return std::make_shared<MiniaudioSource>(mEngine);
+    }
+
+    Listener listener(size_t index) override
+    {
+        if (index >= mListeners.size())
+        {
+            CUBOS_ERROR("Listener index out of range");
+            return nullptr;
+        }
+        return mListeners[index];
+    }
+
+    bool isValid() const
+    {
+        return mValid;
+    }
+
+private:
+    ma_context mContext;
+    ma_device mDevice;
+    ma_engine mEngine;
+    std::vector<std::shared_ptr<MiniaudioListener>> mListeners;
+    bool mValid = false;
+};
+
+MiniaudioContext::MiniaudioContext()
+{
+    if (ma_context_init(nullptr, 0, nullptr, &mContext) != MA_SUCCESS)
+    {
+        CUBOS_FAIL("Failed to initialize audio context.");
+        return;
+    }
+}
+
+MiniaudioContext::~MiniaudioContext()
+{
+    ma_context_uninit(&mContext);
+}
+
+std::string MiniaudioContext::getDefaultDevice()
+{
+    // Create a temporary context
+    ma_context context;
+    if (ma_context_init(nullptr, 0, nullptr, &context) != MA_SUCCESS)
+    {
+        CUBOS_ERROR("Failed to initialize audio context for default device lookup");
+        return "";
+    }
```

Comment:
This method should not be static, so that you don't need to create a temporary context.
You can make it a normal method and then use `mContext`.